### PR TITLE
Adds regenerate_organs proc and adds it to relevant regeneration code

### LINKED ISSUE #21408: ling stasis does not regenerate internal organs.
Be a Ling, loose lungs and die. Revive and you still have no lungs, you can't speak and die of oxygen damage.

### LINKED ISSUE #22734: There is no restore organs function
Turns out issue #21408 is way more general than originally thought. There is NO restore organs function anywhere in the code. This means no type of healing restores organs and the only way to regain organs is through surgery.

This means organs are not restored for:

Changelings using stasis or regenerate. 
healing staffs.
probably admin heals as well?
Anything else you would expect to heal organs.

### PR DESCRIPTION
**Pull request by TehZombehz**
Adds a regenerate_organs proc and adds this proc to relevant regeneration-related code, including:
Wand of healing
Changeling revive and regenerate
Admin revive

Fixes #22734
Fixes #21408

Tell me if I missed anything that needs this.
I threw this together kind of fast, but it was a simple fix. Tell me if I've messed something up in the procs.

### CODE REVIEW

File: code/modules/surgery/bodyparts/dismemberment.dm [hunk 1]
```diff
@@ -311,6 +311,30 @@
 	for(var/Z in limb_list)
 		. += regenerate_limb(Z, noheal)
 
+/mob/living/proc/regenerate_organs()
```
Comment (PKPenguin321) [inline code/modules/surgery/bodyparts/dismemberment.dm]: not sure if this belongs in this file but I can't think of a better place for it so whatever

**Comment by PKPenguin321** [comment]
This is a fix

File: code/modules/surgery/bodyparts/dismemberment.dm (hunk 1 above)
Comment (TehZombehz) [inline code/modules/surgery/bodyparts/dismemberment.dm]: I had the same thought, honestly. If I had to give any reasoning, regenerate_limbs is in the same file and has a similar purpose to regenerate_organs.

**Comment by SgtxRootbeer** [comment]
Does this apply to cortical borer revive?

**Comment by TehZombehz** [comment]
Cortical borer revive doesn't regenerate limbs, so it won't regenerate organs, either.

File: code/modules/surgery/bodyparts/dismemberment.dm (hunk 1 above)
Comment (RemieRichards) [inline code/modules/surgery/bodyparts/dismemberment.dm]: put it in the organ file, not the limbs file, two different concepts these days.

**Comment by Firecage** [comment]
Well....personally I wouldn't mind if admin heal and wands/staffs of healing also gave the mob their brain back. Since it is kinda annoying when you admin heal a mob which you decapitated, only to find they are still dead due to a missing brain.

**Comment by Daturix** [comment]
Should apply with virus regeneration factor. as it is said by name.

**Pull request merged**